doc/BGP/BGP-router-id.md

```diff
@@ -0,0 +1,92 @@
+# BGP Router ID Customized
+
+- [Revision](#revision)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Scope](#scope)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [High Level Design](#high-level-design)
+- [Config DB Enhancement](#config-db-enhancement)
+    - [DEVICE_METADATA](#device_metadata)
+
+### Revision
+
+| Revision | Date        | Author           | Change Description |
+| -------- | ----------- | ---------------- | ------------------ |
+| 1.0      | Mar 27 2024 | Yaqiang Zhu, Jing Kan | Initial proposal   |
+
+### Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+| ------------------------ | ----------- |
+| FRR | A free and open source Internet routing protocol suite for Linux and Unix platforms |
+| Router ID | 32-bit value that uniquely identifies a BGP device |
+
+### Scope
+
+This document describes a mechanism that to use customized router id when Loopback IPv4 address is not configured.
+
+### Overview
+
+Currently, there is hard coding in SONiC to use Loopback IPv4 address to be BGP router id [bgp.main.conf.j2](https://github.com/sonic-net/sonic-buildimage/blob/master/dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2#L91), which would cause BGP cannot work when IPv4 address in Loopback doesn't configure.
```

Comment:
BGP router id was defined as a 32-bit value that uniquely identifies a BGP device. Currently, SONiC uses **Loopback0** IPv4 address as BGP router id. This coupling prevents users from using customized router id.